updater/lib/dependabot/updater/operations/group_update_all_versions.rb

```diff
@@ -105,6 +109,12 @@ def run_ungrouped_dependency_updates
           ).perform
         end
 
+        def pr_exists_for_dependency_group?(group)
+          job.existing_group_pull_requests.
+            each&.
+            any? { |pr| pr.dig(:dependency_group, "name") == group.name }
```

Comment:
I think this `each&` is extra—could be simplified to `job.existing_group_pull_requests.any? { |pr| ...`

Good callout about the hash keys! (One of ruby's warts IMO.) I _think_ only the top-level hash keys will be symbols and the nested ones will be strings. According to the [serializer in dependabot-api](https://github.com/github/dependabot-api/blob/87a3b46341558caa2bead51472a4d0b84932aba4/app/serializers/update_job_serializer_private.rb#L106-L116), I believe this would need to be `pr.dig("dependency_group_name")`, though.